lib/Eclair/AST/Lower.hs

```diff
@@ -28,10 +28,10 @@ compileToRA ast = RA.Module $ concatMap processDecls sortedDecls where
 
   scc :: AST -> [[AST]]
   scc = \case
-    Module _ decls -> map G.flattenSCC sortedDecls
+    Module _ decls -> map G.flattenSCC sortedDecls'
       where
         relevantDecls = filter isRuleOrAtom decls
-        sortedDecls = G.stronglyConnComp $ zipWith (\i d -> (d, i, refersTo d)) [0..] relevantDecls
```

Comment:
Oh wow, this is a nasty case of shadowing. Good that this is fixed now.

(I checked this change myself manually just now, the typechecker would've caught it, but still..)